analyzers/src/SonarAnalyzer.Common/SymbolicExecution/Roslyn/RuleChecks/PublicMethodArgumentsShouldBeCheckedForNullBase.cs

```diff
@@ -34,23 +35,39 @@ protected override ProgramState PreProcessSimple(SymbolicContext context)
     {
         if (NullDereferenceCandidate(context.Operation.Instance) is { } candidate
             && candidate.Kind == OperationKindEx.ParameterReference
-            && candidate.ToParameterReference() is { Parameter: { Type.IsValueType: false } parameter } parameterReference
-            && !HasObjectConstraint(parameter)
-            && !context.CapturedVariables.Contains(parameter) // Workaround to avoid FPs. Can be removed once captures are properly handled by lva.LiveOut()
-            && !parameter.HasAttribute(KnownType.Microsoft_AspNetCore_Mvc_FromServicesAttribute))
+            && candidate.ToParameterReference() is var reference
+            && !reference.Parameter.Type.IsValueType
+            && MissesObjectConstraint(context.State[reference.Parameter])
+            && !reference.Parameter.HasAttribute(KnownType.Microsoft_AspNetCore_Mvc_FromServicesAttribute))
         {
             var message = IsInConstructorInitializer(context.Operation.Instance.Syntax)
                 ? "Refactor this constructor to avoid using members of parameter '{0}' because it could be {1}."
                 : "Refactor this method to add validation of parameter '{0}' before using it.";
-            ReportIssue(parameterReference.WrappedOperation, string.Format(message, parameterReference.WrappedOperation.Syntax, NullName), context);
+            ReportIssue(reference.WrappedOperation, string.Format(message, reference.WrappedOperation.Syntax, NullName), context);
```

Comment:
Is this intentional because of the keyword escaping (e.g. "@event")? We had IParametrSymbol.Name here before.

Reply:
Yes, I was unable to find the name in symbol information, so I reverted to using the syntax tree.